### PR TITLE
fix: Error handling fix for JSONDecodeError on adapters

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Error handling improvement for vertex AI adapter's test connection
- Minor code refactor in `connector_processor.py`

## Why

- Actual root cause was not properly displayed

## How

- Checked all usages of `json.loads()` within the adapters package and noticed that this will help handle it until the adapter / SDK changes are rolled out

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, error handling improvements only

## Notes on Testing

- Checked basic scenario of using an incorrect JSON for the credential

## Screenshots
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/adapter_processor/adapter_processor.py [hunk 1]
```diff
@@ -97,6 +98,12 @@ def test_adapter(adapter_id: str, adapter_metadata: dict[str, Any]) -> bool:
             test_result: bool = adapter_instance.test_connection()
             logger.info(f"{adapter_id} test result: {test_result}")
             return test_result
+        # HACK: Remove after error is explicitly handled in VertexAI adapter
+        except json.JSONDecodeError:
```
Comment (gaya3-zipstack) [inline backend/adapter_processor/adapter_processor.py:102]: Do we get this exception directly originating from test connection of vertex_ai @chandrasekharan-zipstack ?

**Review by harini-venkataraman — APPROVED** [review]
LGTM

File: backend/adapter_processor/adapter_processor.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/adapter_processor/adapter_processor.py:102]: Yes @gaya3-zipstack

**Review by gaya3-zipstack — APPROVED** [review]
Looks fine. However the adapter PR itself can go in and anyway we will need a version roll over everywhere for @Deepak-Kesavan 's cost calculation. I leave it to you @chandrasekharan-zipstack whether you want to push this in or push the proper [adapters PR](https://github.com/Zipstack/unstract-adapters/pull/63) in.

**Comment by chandrasekharan-zipstack** [comment]
> Looks fine. However the adapter PR itself can go in and anyway we will need a version roll over everywhere for @Deepak-Kesavan 's cost calculation. I leave it to you @chandrasekharan-zipstack whether you want to push this in or push the proper [adapters PR](https://github.com/Zipstack/unstract-adapters/pull/63) in.

Do you know the timelines for this version roll over and the cost calculation related fix?
Since there are some other minor changes in this, we could merge it and remove the HACK section once the rollover is done

**Comment by hari-kuriakose** [comment]
@chandrasekharan-zipstack Let's proceed with this for now, assuming there wouldn't be any conflicts.
cc @gaya3-zipstack

**Pull request merged**